engine/src/main/java/org/hibernate/search/spi/SearchIntegratorBuilder.java

```diff
@@ -499,37 +497,6 @@ private static IndexingMode defineIndexingMode(SearchConfiguration cfg) {
 		return IndexingMode.fromExternalRepresentation( indexingStrategy );
 	}
 
-	private ErrorHandler createErrorHandler(SearchConfiguration searchConfiguration) {
-		Object configuredErrorHandler = searchConfiguration.getProperties().get( Environment.ERROR_HANDLER );
```

Comment:
we could simply invoke the `searchConfiguration.getErrorHandler()` here, and if it's not-null give priority to that one.

Ideally, if the ERROR_HANDLER property is also defined log a warning to let people know that it's going to be ignored.

As an alternative strategy you could convert it into a Service, which means you can treat it as a ProvidedService when you need to inject one from Infinispan.